server/src/main/java/org/candlepin/policy/js/pool/PoolRules.java

```diff
@@ -148,32 +148,48 @@ private Pool createBonusPool(Pool masterPool, List<Pool> existingPools) {
 
         log.info("Checking if bonus pools need to be created for pool: {}", masterPool);
 
-        if (attributes.containsKey("virt_limit") && !hasBonusPool(existingPools) && virtQuantity != null) {
-
-            boolean hostLimited = attributes.containsKey("host_limited") &&
-                attributes.get("host_limited").equals("true");
-            HashMap<String, String> virtAttributes = new HashMap<String, String>();
-            virtAttributes.put("virt_only", "true");
-            virtAttributes.put("pool_derived", "true");
-            virtAttributes.put("physical_only", "false");
-            if (hostLimited || config.getBoolean(ConfigProperties.STANDALONE)) {
-                virtAttributes.put("unmapped_guests_only", "true");
-            }
-            // Make sure the virt pool does not have a virt_limit,
-            // otherwise this will recurse infinitely
-            virtAttributes.put("virt_limit", "0");
+        if (attributes.containsKey("virt_limit")  && virtQuantity != null) {
+            if (!hasBonusPool(existingPools)) {
```

Comment:
It is better to :

```java
 Pool  bonusPool = getBonusPool(existingPools);

 if(bonusPool == null) {
    //create bonus pool
 } else {
   //update pool if needed
 }
```
this way we:

- iterate through pools only once
- get to clean up the hasBonusPool method